Use smart pointers in NetworkMDNSRegister

achristensen07 opened the pull request:
#### bfba0fe939951284e0856a4f94537aae1c29005a
<pre>
Use smart pointers in NetworkMDNSRegister
<a href="https://bugs.webkit.org/show_bug.cgi?id=262942">https://bugs.webkit.org/show_bug.cgi?id=262942</a>
rdar://116272449

Reviewed by Youenn Fablet.

Use std::unique_ptr that calls DNSServiceRefDeallocate automatically.
Use CheckedRef for m_connection.
The DNSRecordRef on PendingRegistrationRequest wasn&apos;t used.

* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp:
(WebKit::NetworkMDNSRegister::DNSServiceDeallocator::operator() const):
(WebKit::NetworkMDNSRegister::unregisterMDNSNames):
(WebKit::NetworkMDNSRegister::closeAndForgetService):
(WebKit::NetworkMDNSRegister::registerMDNSName):
(WebKit::NetworkMDNSRegister::sessionID const):
(WebKit::NetworkMDNSRegister::~NetworkMDNSRegister): Deleted.
* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.h:

Canonical link: <a href="https://commits.webkit.org/269166@main">https://commits.webkit.org/269166@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4f56710cde2dc3d2d425f5575e324d1034e534bb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21767 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/22009 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22815 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23635 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20149 "Built successfully") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/26228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22298 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/21283 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21995 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/26228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/18860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24488 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/26228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/25997 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/26228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/19926 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23857 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/20396 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19731 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/23949 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2702 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20324 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp:
```diff
@@ -133,10 +124,10 @@ void NetworkMDNSRegister::registerMDNSName(WebCore::ScriptExecutionContextIdenti
 
-    auto* pendingRequest = new PendingRegistrationRequest(m_connection, WTFMove(name), sessionID(), WTFMove(completionHandler));
-    auto* record = &pendingRequest->record;
+    auto* pendingRequest = new PendingRegistrationRequest(m_connection.get(), WTFMove(name), sessionID(), WTFMove(completionHandler));
```
I would tend to use a makeUnique here and release it in the ok code path instead of deleting it in the error cases.

achristensen07 committed:
Use smart pointers in NetworkMDNSRegister
https://bugs.webkit.org/show_bug.cgi?id=262942
rdar://116272449

Reviewed by Youenn Fablet.

Use std::unique_ptr that calls DNSServiceRefDeallocate automatically.
Use CheckedRef for m_connection.
The DNSRecordRef on PendingRegistrationRequest wasn't used.

* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.cpp:
(WebKit::NetworkMDNSRegister::DNSServiceDeallocator::operator() const):
(WebKit::NetworkMDNSRegister::unregisterMDNSNames):
(WebKit::NetworkMDNSRegister::closeAndForgetService):
(WebKit::NetworkMDNSRegister::registerMDNSName):
(WebKit::NetworkMDNSRegister::sessionID const):
(WebKit::NetworkMDNSRegister::~NetworkMDNSRegister): Deleted.
* Source/WebKit/NetworkProcess/webrtc/NetworkMDNSRegister.h:

Canonical link: https://commits.webkit.org/269166@main
